integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/YcloudHybridCloudTest.java

```diff
@@ -0,0 +1,252 @@
+                .when(sdxTestClient.createInternal(), key(sdxInternal))
+                .awaitForFlow(key(sdxInternal))
+                .await(SdxClusterStatusResponse.RUNNING)
+                .then(this::validateRequestParameters)
+                .validate();
+    }
+
+    private String getDefaultSDXBlueprintName() {
+        return commonClusterManagerProperties().getInternalSdxBlueprintName();
+    }
+
+    private EnvironmentTestDto checkEnvIsListedByNameAndParentName(TestContext testContext,
+            EnvironmentTestDto environment,
+            EnvironmentClient environmentClient) {
+        Collection<SimpleEnvironmentResponse> simpleEnvironmentV4Response = environment.getResponseSimpleEnvSet();
+        if (isNull(simpleEnvironmentV4Response)) {
+            throw new TestFailException("Environment list response is missing.");
+        }
+        boolean listed = simpleEnvironmentV4Response.stream()
+                .anyMatch(environmentResponse -> nameEquals(environment, environmentResponse) && parentNameEquals(environment, environmentResponse));
+        if (!listed) {
+            throw new TestFailException("Environment is not listed");
+        }
+        return environment;
+    }
+
+    private boolean nameEquals(EnvironmentTestDto environment, SimpleEnvironmentResponse environmentResponse) {
+        return environment.getName().equals(environmentResponse.getName());
+    }
+
+    private boolean parentNameEquals(EnvironmentTestDto environment, SimpleEnvironmentResponse environmentResponse) {
+        return isNull(environment.getParentEnvironmentName()) && isNull(environmentResponse.getParentEnvironmentName()) ||
+                environment.getParentEnvironmentName().equals(environmentResponse.getParentEnvironmentName());
+    }
+
+    private SdxInternalTestDto validateRequestParameters(TestContext tc, SdxInternalTestDto dto, SdxClient client) {
+        assertNotNull(dto);
+        SdxInternalClusterRequest sdxRequest = dto.getRequest();
+        assertNotNull(sdxRequest);
+        StackV4Request stackRequest = sdxRequest.getStackV4Request();
+        assertNotNull(stackRequest);
+        ClusterV4Request clusterRequest = stackRequest.getCluster();
+        assertNotNull(clusterRequest);
+        ClouderaManagerV4Request cmRequest = clusterRequest.getCm();
+        assertNotNull(cmRequest);
+
+        validateInstanceGroups(stackRequest.getInstanceGroups());
+        validateRepository(cmRequest.getRepository());
+        validateProducts(cmRequest.getProducts());
+
+        return dto;
+    }
+
+    private void validateInstanceGroups(List<InstanceGroupV4Request> instanceGroups) {
+        assertNotNull(instanceGroups);
+        assertEquals("The result instance group list has " + instanceGroups.size() + " items instead of 2", 2, instanceGroups.size());
+        assertTrue(instanceGroups.stream().anyMatch(ig -> ig.getName().equals(MASTER_INSTANCE_GROUP) || ig.getName().equals(IDBROKER_INSTANCE_GROUP)));
```

Comment:
I think you should test that both MASTER and IDBROKER instance groups are available at the same time.